src/main/java/com/vinylteam/vinyl/entity/User.java

```diff
@@ -1,4 +1,71 @@
 package com.vinylteam.vinyl.entity;
 
+import java.util.Base64;
+
 public class User {
+
+    private String email;
+    private String password;
+    private String salt;
+    private int iterations;
+    private Role role;
+
+    public User(String email, String password, byte[] salt, int iterations, Role role) {
+        this.email = email;
+        this.password = password;
+        this.salt = Base64.getEncoder().encodeToString(salt);
+        this.iterations = iterations;
+        this.role = role;
+    }
```

Comment:
Add also constructor without parameters. Or remove all constructors and leave only default.
Relocate operations with coding and encoding to security service.